WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantActivity.kt

```diff
@@ -38,22 +55,46 @@ class AuthDeviceGrantActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_auth)
-        val viewModel by viewModels<AuthDeviceGrantViewModel>()
-
-        // Start the OAuth flow when the user presses the button
-        findViewById<View>(R.id.authenticateButton).setOnClickListener {
-            viewModel.startAuthFlow()
-        }
-
-        // Show current status on the screen
-        viewModel.status.observe(this) { statusText ->
-            findViewById<TextView>(R.id.status_text_view).text = resources.getText(statusText)
-        }
+        setContent { AuthenticateApp(deviceGrantViewModel = viewModel()) }
+    }
+}
 
-        // Show dynamic content on the screen
-        viewModel.result.observe(this) { resultText ->
-            findViewById<TextView>(R.id.result_text_view).text = resultText
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun AuthenticateApp(deviceGrantViewModel: AuthDeviceGrantViewModel) {
+    AppScaffold {
+        val uiState = deviceGrantViewModel.uiState.collectAsState()
+        val localContext = LocalContext.current
+        val columnState = rememberResponsiveColumnState(
+            contentPadding = ScalingLazyColumnDefaults.padding(
+                first = ItemType.Unspecified,
```

Comment:
with the content below, this should be `first = Text` & `last = Text`

Reply:
Done.

Reply:
Is it though?